temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -145,23 +162,46 @@ public TestWorkflowService(boolean lockTimeSkipping) {
 
   // TODO: Shutdown.
   public TestWorkflowService() {
-    serverName = InProcessServerBuilder.generateName();
+    client = new Client();
+    try {
+      InProcessServerBuilder.forName(client.serverName)
+          .directExecutor()
+          .addService(this)
+          .build()
+          .start();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    outOfProcessServer = null;
+  }
+
+  // Creates an out-of-process rather than in-process server, and does not set up a client.
+  public TestWorkflowService(int port) {
+    log.info("Server started, listening on " + port);
+    client = null;
```

Comment:
All functions that are using client like `newClientStub` and `close` will now thrown an NPE.

Reply:
Hmm, on the server side those functions shouldn't ever be called.  I'll hide this constructor and make a factory which clarifies the scenario when this constructor should be used.